#! /bin/sh /usr/share/dpatch/dpatch-run
## no_loopback_checksum.dpatch by  <apollock@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Deal with the fact that the loopback interface isn't checsummed by Linux

@DPATCH@
diff -urNad isc-dhcp~/common/packet.c isc-dhcp/common/packet.c
--- isc-dhcp~/common/packet.c	2009-06-23 22:15:20.000000000 -0700
+++ isc-dhcp/common/packet.c	2009-08-31 21:40:19.000000000 -0700
@@ -319,6 +319,11 @@
 				  checksum((unsigned char *)&ip.ip_src,
 					   8, IPPROTO_UDP + ulen))));
 
+  /* loopback interface isn't checksumed in linux */
+  if (interface -> hw_address.hbuf [0] == 0) {
+    usum = 0;
+  }
+
   udp_packets_seen++;
   if (usum && usum != sum) {
 	  udp_packets_bad_checksum++;
